Enigmatry.Entry.AspNetCore.Tests/SampleAppTests/WeatherForecastControllerFixtureBase.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Net;
+using Enigmatry.Entry.AspNetCore.SampleApp;
+using FluentAssertions;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Enigmatry.Entry.AspNetCore.Tests.SampleAppTests;
+
+[Category("integration")]
+public abstract class WeatherForecastControllerFixtureBase : SampleAppFixtureBase
+{
+    private readonly bool _newtonsoftJsonUsed;
+
+    protected WeatherForecastControllerFixtureBase(SampleAppSettings settings) : base(settings)
+    {
+        _newtonsoftJsonUsed = settings.UseNewtonsoftJson;
+    }
+
+    protected abstract Task<T?> GetAsync<T>(HttpClient client, string uri);
+
+    protected abstract T? DeserializeJson<T>(string content);
+
+    [Test]
+    public async Task TestGet()
+    {
+        var response = await GetAsync<WeatherForecast[]>(Client, "WeatherForecast");
+        response!.Length.Should().Be(5);
+    }
+
+    [Test]
+    public async Task TestGetError()
+    {
+        var response = await Client.GetAsync("WeatherForecast/ThrowsError");
+
+        response.StatusCode.Should().Be(HttpStatusCode.InternalServerError);
+        response.Content.Headers.ContentType!.Should().BeNull();
+    }
+
+    [Test]
+    public async Task TestGetNotFound()
+    {
+        HttpResponseMessage response = await Client.GetAsync("WeatherForecast/NotFound");
+
+        response.StatusCode.Should().Be(HttpStatusCode.NotFound);
+        response.Content.Headers.ContentType!.Should().BeNull();
+    }
+
+    [Test]
+    public async Task TestGetValidationError()
+    {
+        var response = await Client.GetAsync("WeatherForecast/ProblemDetails");
+
+        response.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+        response.Content.Headers.ContentType!.MediaType.Should().Be("application/problem+json");
+
```

Comment:
This test isn't really following AAA best practices. Assert part should be like 1-3 lines of code.
Once you deserialize ProblemDetails, you can compare it with an object instead of comparing every individual property. One step further would be to use a library (verify.nunit) to compare result with an object from given file.

Reply:
Ok, I'll dive deeper into this since Verify can be useful for me not just on this.